### ISSUE TITLE
loss is nan;rpn_box: is nan;why?

### ISSUE THREAD

**Issue by lixiaorui4444** [jwyang/faster-rcnn.pytorch#562]
Loading pretrained weights from data/pretrained_model/resnet101_caffe.pth
[session 1][epoch  1][iter    0/2892] loss: 2.0858, lr: 1.00e-04
                        fg/bg=(10/118), time cost: 0.233441
                        rpn_cls: 0.7161, rpn_box: 0.0311, rcnn_cls: 1.1516, rcnn_box 0.1871
[session 1][epoch  1][iter  100/2892] loss: nan, lr: 1.00e-04
                        fg/bg=(7/121), time cost: 19.324604
                        rpn_cls: 0.6902, rpn_box: nan, rcnn_cls: 0.2591, rcnn_box 0.0000
[session 1][epoch  1][iter  200/2892] loss: nan, lr: 1.00e-04
                        fg/bg=(128/0), time cost: 19.772832
                        rpn_cls: 0.6851, rpn_box: nan, rcnn_cls: 1.0205, rcnn_box 0.0000

hi everyone, I get a problem.And I have read the issues that had the same problem, but any of them could not help me. Could anyone who help me? Thanks lot.

**Comment by lixiaorui4444** [comment]
And I only change the self._classes  in class pascal_voc as following:
self._classes = ('__background__',  # always index 0
                        'hat','head')

**Comment by lixiaorui4444** [comment]
There are some samples that are wrong, I have got it. If any question, please contact to me qq:1664010627

**Comment by MatthewInkawhich** [comment]
@lixiaorui4444 I am having the same issue trying to run on my own dataset. Note that I have already removed the **-1** from the pascal_voc.py file to avoid integer underflow.

What exactly do you mean by "There are some samples that are wrong"?

Thanks!

**Comment by lixiaorui4444** [comment]
some samples in  my database are wrong. Usually, we need the coordinate of left upper corner and upper right corner. But the wrong samples give other point coordinate such as right upper corner. I am not good at English. If you can understand Chinese, the following will help you understand.
我的数据库的样本里面有几个坐标是错误的，通常我们都是需要左上角和右下角的坐标，但是我的样本那里面有30个左右的图片，提供给我的坐标不是左上角 和右下角，导致产生了nan的结果，把这些错误的样本剔除之后，就可以完美的训练了，希望可以帮到你。

**Comment by lixiaorui4444** [comment]
@MatthewInkawhich

**Comment by Tuich** [comment]
@lixiaorui4444 你好，请问您遇到的数据集标注错误具体是什么情况？标注有小于0的地方吗？
还是只是bbox没有标对？如果是这样为什么数据集标注错误会造成loss=nan？

**Comment by lixiaorui4444** [comment]
> @lixiaorui4444 你好，请问您遇到的数据集标注错误具体是什么情况？标注有小于0的地方吗？
> 还是只是bbox没有标对？如果是这样为什么数据集标注错误会造成loss=nan？

我是因为样本标签坐标有问题，不清楚你说的小于0是什么问题

**Comment by Tuich** [comment]
@lixiaorui4444 非常感谢您的回答！请问您是怎么找出数据集中的标注错误的呢？我的数据集有2500张图片，我不太想手动找。
其实我还有一个问题：为什么错误的标注会导致loss为nan？进而不能训练呢？错了就错了呗，其他batch为什么会受到影响？
